### PR TITLE
Updated references to camera2 android feature

### PR DESCRIPTION
**Pull request by gonzalonm**
Updated on Camera2Basic

### CODE REVIEW

**Comment by gonzalonm** [comment]
Bug: https://bugzilla.xamarin.com/show_bug.cgi?id=42390

File: android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs [hunk 1]
```diff
@@ -4,524 +4,1050 @@
...
+            {
+                return true;
+            }
+
+            public void OnSurfaceTextureSizeChanged(Android.Graphics.SurfaceTexture surface, int width, int height)
+            {
+                owner.ConfigureTransform(width, height);
+            }
+
+            public void OnSurfaceTextureUpdated(Android.Graphics.SurfaceTexture surface)
```
Comment (gonzalonm) [inline android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs]: It belongs to interface `TextureView.ISurfaceTextureListener`. So, it forces you to implement something in.

File: android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs [hunk 2]
```diff
@@ -4,524 +4,1050 @@
...
+        private static readonly int MAX_PREVIEW_WIDTH = 1920;
+
+        // Max preview height that is guaranteed by Camera2 API
+        private static readonly int MAX_PREVIEW_HEIGHT = 1080;
+
+        // TextureView.ISurfaceTextureListener handles several lifecycle events on a TextureView
+        private Camera2BasicSurfaceTextureListener mSurfaceTextureListener;
+        private class Camera2BasicSurfaceTextureListener : Java.Lang.Object, TextureView.ISurfaceTextureListener
+        {
+            public Camera2BasicFragment owner;
```
Comment (gonzalonm) [inline android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs]: Done

File: android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs [hunk 3]
```diff
@@ -4,524 +4,1050 @@
...
+            public override void OnCaptureProgressed(CameraCaptureSession session, CaptureRequest request, CaptureResult partialResult)
+            {
+                Process(partialResult);
+            }
+
+            private void Process(CaptureResult result)
+            {
+                switch (owner.mState)
+                {
+                    case STATE_PREVIEW:
```
Comment (gonzalonm) [inline android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs]: Done

File: android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs [hunk 4]
```diff
@@ -4,524 +4,1050 @@
...
+            else
+            {
+                FragmentCompat.RequestPermissions(this, new string[] { Manifest.Permission.Camera },
+                        REQUEST_CAMERA_PERMISSION);
+            }
+        }
+
+        public void OnRequestPermissionsResult(int requestCode, string[] permissions, int[] grantResults)
+        {
+            if (requestCode == REQUEST_CAMERA_PERMISSION)
```
Comment (gonzalonm) [inline android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs]: Done

File: android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs [hunk 5]
```diff
@@ -4,524 +4,1050 @@
...
+            {
+                if (grantResults.Length != 1 || grantResults[0] != (int)Permission.Granted)
+                {
+                    ErrorDialog.NewInstance(GetString(Resource.String.request_permission))
+                            .Show(ChildFragmentManager, FRAGMENT_DIALOG);
+                }
+            }
+            else
```
Comment (gonzalonm) [inline android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs]: Done

File: android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs [hunk 6]
```diff
@@ -4,524 +4,1050 @@
...
+            {
+                public void OnClick(IDialogInterface dialog, int which)
+                {
+                    mActivity.Finish();
+                }
+            }
+
+            public static ErrorDialog NewInstance(string message)
+            {
+                ErrorDialog dialog = new ErrorDialog();
```
Comment (gonzalonm) [inline android5.0/Camera2Basic/Camera2Basic/Camera2BasicFragment.cs]: Done

**Comment by gonzalonm** [comment]
@olegoid I've made changes based on your feedback. Please, take a look them and let me know any changes to do.

**Comment by olegoid** [comment]
Looks pretty good! However, I have one more request ;)

Could you put classes in defined in `Camera2BasicFragment.cs` to separate files. Now `Camera2BasicFragment.cs` has more than 1000 lines and it's pretty hard to deal with.

**Comment by gonzalonm** [comment]
Updated. Now `Camera2BasicFragment.cs` has almost 700 lines. Let me know how it looks.

**Comment by olegoid** [comment]
LGTM

**Pull request merged**